### ISSUE TITLE
Gradient in distributed training

### ISSUE THREAD

**Issue by KinglittleQ** [KevinMusgrave/pytorch-metric-learning#363]
Here is the current implementation of the distributed wrapper.

https://github.com/KevinMusgrave/pytorch-metric-learning/blob/6bfa880b8d2acafb7c6d52041d2bb14ed41aee76/src/pytorch_metric_learning/utils/distributed.py#L35-L59

But according to this [blog](https://amsword.medium.com/gradient-backpropagation-with-torch-distributed-all-gather-9f3941a381f8), the loss function should be multiplied by `world_size`. 

What do you think about it? Maybe I can create a PR to fix it.

**Comment by KevinMusgrave** [comment]
Yeah I guess you're right. If you create a PR, please work off of the "dev" branch.

**Comment by KevinMusgrave** [comment]
Reopening until its in the main branch
